recipes-kernel/linux/linux-adi/0001-sc59x-fastautomdix-support.patch

```diff
@@ -0,0 +1,98 @@
+diff --git a/arch/arm/boot/dts/sc594-som-ezkit.dts b/arch/arm/boot/dts/sc594-som-ezkit.dts
+index 25acefc408c2..fd3d43c8eaa6 100644
+--- a/arch/arm/boot/dts/sc594-som-ezkit.dts
++++ b/arch/arm/boot/dts/sc594-som-ezkit.dts
+@@ -51,24 +51,22 @@
+ 			              <&ssw0 4 GPIO_ACTIVE_LOW>;	/* SPI2D2_D3_EN */
+ 		};
+ 
+-/* The button GPIO conflicts with OSPI0_D4 for this
++/* The button GPIO conflicts with OSPI0_D4 for this */
+ 		button0: button@0 {
+ 			compatible = "adi,button-led";
+-			en-pins = <&ssw1 1 GPIO_ACTIVE_LOW>;	/* PUSHBUTTON_EN
+-			button_gpio = <48>; /* PD_00
+-			led_gpio = <35>; /* PC_03 / LED9 / LED1_KIT
++			en-pins = <&ssw1 1 GPIO_ACTIVE_LOW>;	/* PUSHBUTTON_EN */
++			button_gpio = <50>; /* PD_00 */
++			led_gpio = <35>; /* PC_03 / LED9 / LED1_KIT */
+ 		};
+-*/
+ 
+ /* This also appears to interfere with the OSPI somehow...
+-   possible trace noise/interference?
++/*  possible trace noise/interference? */
+ 		button1: button@1 {
+ 			compatible = "adi,button-led";
+-			en-pins = <&ssw1 1 GPIO_ACTIVE_HIGH>; /* PUSHBUTTON_EN
+-			button_gpio = <112>; /* PH_00
+-			led_gpio = <34>; /* PC_02 / LED8 / LED2_KIT
++			en-pins = <&ssw1 1 GPIO_ACTIVE_HIGH>; /* PUSHBUTTON_EN */
++			button_gpio = <112>; /* PH_00 */
++			led_gpio = <34>; /* PC_02 / LED8 / LED2_KIT */ 
+ 		};
+-*/
+ 
+ 		core1-rproc@0x3108c000 {
+ 			compatible = "adi,remoteproc";
+diff --git a/drivers/net/phy/dp83867.c b/drivers/net/phy/dp83867.c
+index 3aa1ae211f5f..61f50504f6c6 100644
+--- a/drivers/net/phy/dp83867.c
++++ b/drivers/net/phy/dp83867.c
+@@ -23,6 +23,8 @@
+ #define MII_DP83867_ISR		0x13
+ #define DP83867_CTRL		0x1f
+ #define DP83867_CFG3		0x1e
++#define PHY_CFG3_FASTLINK       0x8000
++#define PHY_CFG3_FAST_AUT_MDIX  0x0100
+ 
+ /* Extended Registers */
+ #define DP83867_CFG4            0x0031
+@@ -424,7 +426,13 @@ static int dp83867_config_init(struct phy_device *phydev)
+ 		phy_write(phydev, DP83867_CFG3, val);
+ #if (!defined(CONFIG_ARCH_SC58X)) && (!defined(CONFIG_ARCH_SC57X))
+ 	}
++         /*PHY FAST Link up - CFG3*/
++        val = phy_read(phydev, DP83867_CFG3);
++        val |= (PHY_CFG3_FASTLINK | PHY_CFG3_FAST_AUT_MDIX);
++        phy_write(phydev, DP83867_CFG3, val);
+ #endif
++
++	
+ 	if (dp83867->port_mirroring != DP83867_PORT_MIRROING_KEEP)
+ 		dp83867_config_port_mirroring(phydev);
+ 
+diff --git a/drivers/net/phy/phy.c b/drivers/net/phy/phy.c
+index 105d389b58e7..c75b9e8a13a1 100644
+--- a/drivers/net/phy/phy.c
++++ b/drivers/net/phy/phy.c
+@@ -952,7 +952,7 @@ void phy_state_machine(struct work_struct *work)
+ 	 */
+ 	mutex_lock(&phydev->lock);
+ 	if (phy_polling_mode(phydev) && phy_is_started(phydev))
+-		phy_queue_state_machine(phydev, PHY_STATE_TIME);
++		phy_queue_state_machine(phydev, PHY_STATE_TIME/50);
```

Comment:
How much are we setting the polling period here in ms ?

Reply:
20msec is polling interval. Added comment mentioning it